### PR TITLE
BookController and dto implemented

### PR DESCRIPTION
**Pull request by jszymanski97**
BookController.java - created with supporting BookDto.java and CreateBookRequestDto.java classes
MapperConfig.java - implemented mapstruct to ease the class conversion
EntityNotFoundException.java - exception created for error message clarity

### CODE REVIEW

File: src/main/java/mate/project/dto/BookDto.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package mate.project.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookDto {
+
```
Comment (MateuszRostek) [inline src/main/java/mate/project/dto/BookDto.java]: Do not add redundant empty spaces.

File: src/main/java/mate/project/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package mate.project.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+
```
Comment (MateuszRostek) [inline src/main/java/mate/project/dto/CreateBookRequestDto.java]: Do not add redundant empty spaces.

File: src/main/java/mate/project/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+}
```
Comment (MateuszRostek) [inline src/main/java/mate/project/dto/CreateBookRequestDto.java]: Make it a `record`.

File: src/main/java/mate/project/mapper/BookMapper.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import mate.project.config.MapperConfig;
+import mate.project.dto.BookDto;
+import mate.project.dto.CreateBookRequestDto;
+import mate.project.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+
```
Comment (MateuszRostek) [inline src/main/java/mate/project/mapper/BookMapper.java]: Do not add redundant empty spaces.

File: src/main/java/mate/project/repository/BookRepository.java [hunk 5]
```diff
@@ -7,4 +7,6 @@ public interface BookRepository {
     Book save(Book book);
 
     List<Book> findAll();
+
+    Book getBookById(Long id);
```
Comment (MateuszRostek) [inline src/main/java/mate/project/repository/BookRepository.java]: You should use Optional here:
```suggestion
    Optional<Book> getBookById(Long id);
```

File: src/main/java/mate/project/service/BookServiceImpl.java [hunk 6]
```diff
@@ -1,28 +1,38 @@
...
+import mate.project.mapper.BookMapper;
 import mate.project.model.Book;
 import mate.project.repository.BookRepository;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
+
```
Comment (MateuszRostek) [inline src/main/java/mate/project/service/BookServiceImpl.java]: Do not add redundant empty spaces.

File: src/main/java/mate/project/service/BookServiceImpl.java [hunk 7]
```diff
@@ -1,28 +1,38 @@
...
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto createBookRequestDto) {
+        Book model = bookMapper.toModel(createBookRequestDto);
+        Book savedModel = bookRepository.save(model);
+        return bookMapper.toBookDto(savedModel);
```
Comment (MateuszRostek) [inline src/main/java/mate/project/service/BookServiceImpl.java]:
```suggestion
        return bookMapper.toBookDto(bookRepository.save(model));
```

File: src/main/resources/application.properties
Comment (MateuszRostek) [inline src/main/resources/application.properties:1]: Add this one also:

`spring.jpa.open-in-view=false`

**Review by MateuszRostek — CHANGES_REQUESTED** [review]
Nice try! See some advice and implement required changes.

File: src/main/java/mate/project/repository/BookRepository.java (hunk 5 above)
Comment (jszymanski97) [reply src/main/java/mate/project/repository/BookRepository.java]: Okay, I went by the video where the coach explained to use optional for methods like findBookById where we not sure if the book exists, and for methods with get we dont need to. But I see and understand why it is better to use optional for both cases

**Comment by MateuszRostek** [comment]
In general I do not recommend to use and create methods of type `get`, it is safer to create methods of type `find` and return Optional from them:)

**Review by MateuszRostek — APPROVED** [review]
Good job!

**Comment by MateuszRostek** [comment]
Remember to merge this branch and create next solution on the new one with new pull request:)

**Pull request merged**